coordinator/coordinator.go

```diff
@@ -16,11 +18,93 @@ import (
 	"github.com/golang/protobuf/proto"
 )
 
+type RunningQuery struct {
+	userName             string
+	databaseName         string
+	queryString          string
+	startTime            time.Time
+}
+
+type RunningQueryList  []*RunningQuery
+type RunningQueries struct {
+	data                 map[*RunningQuery]struct{}
+	lock                 sync.Mutex
+}
+
 type Coordinator struct {
 	clusterConfiguration *cluster.ClusterConfiguration
 	raftServer           *RaftServer
 	config               *configuration.Configuration
 	permissions          Permissions
+	runningQueries       *RunningQueries
+}
+
+func NewRunningQuery(
+	userName string, databaseName string, queryString string, startTime time.Time) *RunningQuery {
+	runningQuery := &RunningQuery{
+		userName:             userName,
+		databaseName:         databaseName,
+		queryString:          queryString,
+		startTime:            startTime,
+	}
+
+	return runningQuery
+}
+
+func NewRunningQueries() *RunningQueries {
+	runningQueries := &RunningQueries{
+		data:                 make(map[*RunningQuery]struct {}),
+	}
+
+	return runningQueries
```

Comment:
I followed what it look like a "pattern". 
See eg. NewCoordinator here: https://github.com/Wikia/influxdb/pull/1/files#diff-cd24963c47c3445deb5ec946ac538602R110